openstf/tasks/create_components_forecast.py

```diff
@@ -29,21 +29,78 @@
 
 
 """
-from openstf.pipeline.create_forecast import make_components_prediction
+import structlog
+from datetime import datetime, timedelta
+
+from openstf.pipeline.create_component_forecast import (
+    create_components_forecast_pipeline,
+)
 from openstf.tasks.utils.utils import check_status_change, update_status_change
 from openstf.tasks.utils.predictionjobloop import PredictionJobLoop
 from openstf.tasks.utils.taskcontext import TaskContext
 
+T_BEHIND_DAYS = 0
+T_AHEAD_DAYS = 3
+
 
-def create_components_forecast_pj(pj, context):
+def create_components_forecast_task(pj, context):
+    """Top level task that creates a components forecast.
+    On this task level all database and context manager dependencies are resolved.
+
+    Args:
+        pj (dict): Prediction job
+        context (TaskContext): Contect object that holds a config manager and a database connection
+    """
+    logger = structlog.get_logger(__name__)
     if pj["train_components"] == 0:
         context.logger.info(
             "Skip prediction job", train_components=pj["train_components"]
         )
         return
 
+    # Define datetime range for input data
+    datetime_start = datetime.utcnow() - timedelta(days=T_BEHIND_DAYS)
+    datetime_end = datetime.utcnow() + timedelta(days=T_AHEAD_DAYS)
+
+    logger.info(
+        "Get predicted load", datetime_start=datetime_start, datetime_end=datetime_end
+    )
+    # Get most recent load forecast
+    input_data = context.database.get_predicted_load(
```

Comment:
hmm. this seems confusing as well. We get something which we saved as a **forecast** by calling **get_predicted_load** and call it **input_data** and commented it with **load forecast**. Perhaps its possible to be a bit more consistent.